### PR TITLE
PLAT-1219 DbForeignFieldTest: Properly reimplement prefetching-related tests

### PR DESCRIPTION
**Pull request by alexander-schmidt**

### CODE REVIEW

File: platform-db/src/test/java/com/softicar/platform/db/runtime/field/DbForeignFieldTest.java [hunk 1]
```diff
@@ -118,43 +117,91 @@ public void testPrefetchingNullReferences() {
...
+		assertTrue(object.invalidated());
+		assertFalse(object.stub());
+		assertFalse(referencingObject.invalidated());
+		assertFalse(referencingObject.stub());
+
+		// execute: prefetch the referencing object
+		DbTestObject.FOREIGN_FIELD.prefetch(List.of(referencingObject));
+
+		// assert result
+		assertEquals("updated", referencingObject.getForeign().getString());
```
Comment (oliver-richers) [inline platform-db/src/test/java/com/softicar/platform/db/runtime/field/DbForeignFieldTest.java:145]: actually, the call to `getString` de-invalidates the object in any case, so we don't test the prefeching in any way
```suggestion
		assertFalse(object.invalidated());
		assertEquals("updated", DbTestObject.STRING_FIELD.getValueDirectly(object));
```

File: platform-db/src/test/java/com/softicar/platform/db/runtime/field/DbForeignFieldTest.java [hunk 2]
```diff
@@ -118,43 +117,91 @@ public void testPrefetchingNullReferences() {
...
+		assertFalse(object.invalidated());
+		assertTrue(object.stub());
+		assertFalse(referencingObject.invalidated());
+		assertTrue(referencingObject.stub());
+
+		// execute: prefetch the referencing object
+		DbTestObject.FOREIGN_FIELD.prefetch(List.of(referencingObject));
+
+		// assert result
+		assertEquals("updated", referencingObject.getForeign().getString());
```
Comment (oliver-richers) [inline platform-db/src/test/java/com/softicar/platform/db/runtime/field/DbForeignFieldTest.java:179]: same as above
```suggestion
		assertFalse(object.stub());
		assertEquals("updated", DbTestObject.STRING_FIELD.getValueDirectly(object));
```

File: platform-db/src/test/java/com/softicar/platform/db/runtime/field/DbForeignFieldTest.java [hunk 3]
```diff
@@ -118,43 +117,91 @@ public void testPrefetchingNullReferences() {
...
+		assertFalse(object.invalidated());
+		assertFalse(object.stub());
+		assertFalse(referencingObject.invalidated());
+		assertFalse(referencingObject.stub());
+
+		// execute: prefetch the referencing object
+		DbTestObject.FOREIGN_FIELD.prefetch(List.of(referencingObject));
+
+		// assert result
+		assertEquals("cached", referencingObject.getForeign().getString());
```
Comment (oliver-richers) [inline platform-db/src/test/java/com/softicar/platform/db/runtime/field/DbForeignFieldTest.java:205]: same as above
```suggestion
		assertEquals("cached", DbTestObject.STRING_FIELD.getValueDirectly(object));
```

**Review by oliver-richers — CHANGES_REQUESTED** [review]

File: platform-db/src/test/java/com/softicar/platform/db/runtime/field/DbForeignFieldTest.java (hunk 1 above)
Comment (alexander-schmidt) [reply platform-db/src/test/java/com/softicar/platform/db/runtime/field/DbForeignFieldTest.java:145]: I see. Quite tough to see. Fixed.

**Review by oliver-richers — APPROVED** [review]

**Pull request merged**